term/src/main/jni/termExec.cpp

```diff
@@ -71,8 +70,8 @@ static void android_os_Exec_setPtyUTF8Mode(JNIEnv *env, jobject clazz, jint fd,
         tios.c_iflag &= ~IUTF8;
     }
 
-    if (my_tcsetattr(fd, TCSANOW, &tios) < 0)
```

Comment:
There is no need, this PR's last commit should fix the issue. You should have passed `TCSETS` to ioctl call instead of `TCSANOW`.